drivers/hmis/app/graphql/types/application/client_access_summary.rb

```diff
@@ -14,6 +14,8 @@ class Application::ClientAccessSummary < Types::BaseObject
     field :client_name, String, null: true
 
     def client_name
+      return client&.masked_name unless current_permission?(permission: :can_view_client_name, entity: client)
+
       client&.brief_name
```

Comment:
@eanders @martha I'm a but unsure on this one (General comment, not just for this line).

Should we support scenarios where:
* User CAN **audit clients**, but CANNOT see client [names | dob | ssn | contact info]
* User CAN **audit users**, but CANNOT see client [names | dob | ssn | contact info]

These use-cases both feel highly unlikely to me.
We already have some code to pull DOB/SSN from audit history to support that first case. I wonder if the extra code complexity is worth it. I am leaning towards pulling out this complexity, and instead updating the audit descriptions to say something like:
> This permission allows the user to view changes to any client details, including DOB, SSN, and name, even if they don't have the specific permissions to view those fields on the Client.

Those audit permissions are already marked as **administrative** so this seem pretty reasonable to me. I would prefer it just for the sake of simpler code.

Reply:
I think that makes sense.  If you are auditing, you can see some raw back-end data, which we don't fully filter.  Alternatively, we could require that if you can audit clients you must also have the other "view" permissions...

Reply:
I don't like that approach as much. For gr staff for example, it may be useful to enable auditing for some specific debugging process, but to leave the overall viewing perms turned _off_ to reduce exposure elsewhere in the application.

Reply:
Yeah, I agree, let's just make it clear what is being exposed if you have the audit permission.

Reply:
Okay sounds good, I agree too - as you can see in the contact info permissions [PR](https://github.com/greenriver/hmis-warehouse/pull/4220#discussion_r1548565518), that change added even more complexity to the audit field permissions.

Reply:
I made this update and just wanted to note that in doing so, I merged in the client contact info permissions (just to avoid a merge conflict) - so this PR now includes new permissions for both name and contact info.

Reply:
sounds good! thanks Martha